functional/test/test_streams.py

```diff
@@ -145,3 +146,91 @@ def test_to_csv(self):
         sequence.to_csv(tmp_path)
         result = seq.csv(tmp_path).to_list()
         self.assertEqual(expect, result)
+
+    def test_to_sqlite3_failure(self):
+        # test failure case
+        insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+        with self.assertRaises(ValueError):
+            seq(elements).to_sqlite3(1, insert_sql)
+
+    def test_to_sqlite3_file(self):
+        tmp_path = 'functional/test/data/tmp/test.db'
+
+        with sqlite3.connect(tmp_path) as conn:
+            conn.execute("DROP TABLE IF EXISTS user;")
+            conn.execute("CREATE TABLE user (id INT, name TEXT);")
+            conn.commit()
+
+        insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+
+        # test insert into a file
+        seq(elements).to_sqlite3(tmp_path, insert_sql)
+        result = seq.sqlite3(tmp_path, "SELECT id, name FROM user;").to_list()
+        self.assertListEqual(elements, result)
+
+
+    def test_to_sqlite3_query(self):
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+
+        # test insert tuples by query
+        with sqlite3.connect(":memory:") as conn:
+            conn.execute("CREATE TABLE user (id INT, name TEXT);")
+            conn.commit()
+
+            insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+            seq(elements).to_sqlite3(conn, insert_sql)
+            result = seq.sqlite3(conn, "SELECT id, name FROM user;").to_list()
+            self.assertListEqual(elements, result)
+
+    def test_to_sqlite3_tuple(self):
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+
+        # test isnert tuples by table name
+        with sqlite3.connect(":memory:") as conn:
+            conn.execute("CREATE TABLE user (id INT, name TEXT);")
+            conn.commit()
+
+            table_name = "user"
+            seq(elements).to_sqlite3(conn, table_name)
+            result = seq.sqlite3(conn, "SELECT id, name FROM user;").to_list()
+            self.assertListEqual(elements, result)
+
+    def test_to_sqlite3_namedtuple(self):
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+        user = collections.namedtuple("user", ["id", "name"])
```

Comment:
Good test here is to make two named tuples, one being `namedtuple('user', ['id', 'name'])` and another `namedtuple('user', ['name', 'id'])` to check that its works regardless of field order